geospaas_processing/downloaders.py

```diff
@@ -230,6 +231,36 @@ def download_file(cls, file, url, connection):
         connection.retrbinary(f"RETR {urlparse(url).path}", file.write)
 
 
+class LocalDownloader(Downloader):
+    """Downloader for locally hosted files, so basically a file copier
+    """
+
+    @staticmethod
+    def get_auth(kwargs):
+        return (None, None)
+
+    @classmethod
+    def connect(cls, url, auth=(None, None)):
+        return None
+
+    @classmethod
+    def close_connection(cls, connection):
+        return None
+
+    @classmethod
+    def get_file_name(cls, url, connection):
+        return os.path.basename(url)
+
+    @classmethod
+    def get_file_size(cls, url, connection, auth=(None, None)):
+        return os.path.getsize(url)
+
+    @classmethod
+    def download_file(cls, file, url, connection):
+        with open(url, 'rb') as source:
+            shutil.copyfileobj(source, file)
```

Comment:
why do you define this methods as classmethods? shouldn't be as a normal object method?

Reply:
Because they are defined like that in the parent class.
And they are like that in the parent class because Downloaders don't use any instance attributes.